js/MetricHandler.js

```diff
@@ -137,6 +137,10 @@ export class MetricHandler {
     return this.metricQHistory.search(inputStr)
   }
 
+  searchMetricsPromiseAndMetadata (inputStr) {
```

Comment:
The name is weird. If we were really to stick with the scheme, it would be something like `searchMetricsWithMetadataPromise`. But I'd rather just use the existing `searchMetricsPromise` and add a `metadata` parameter there that's passed through to `search`.